docs/xks/developer-guide/observability.md

```diff
@@ -0,0 +1,244 @@
+
+### Logging
+
+To gather logs we use the grafana agent operator that deploys a deamonset and runs all the nodes in the cluster and reads the logs from the kubernetes log location.
+Just like metrics you have to define label `xkf.xenit.io/monitoring: tenant` in your PodLogs.
+The PodLogs CRD is created by the grafana agent-operator and functions very similar to how the prometheus operator works especially when it comes to selectors.
+Below you will find a very basic example that will scrape a single pod in the namespace where it's created.
+
+```podlogs.yaml
+apiVersion: monitoring.grafana.com/v1alpha1
+kind: PodLogs
+metadata:
+  name: app1
+  labels:
+    xkf.xenit.io/monitoring: tenant
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: app1
+  pipelineStages:
+    - cri: {}
+```
+
+You can do allot of configuration when it comes to log filtering using PodLogs. For example you can remove drop specific log types that you don't want to send to your long time storage.
+Sadly the grafana agent operator don't supply great documentation around how to define this configuration in the operator.
+But together with running `kubectl explain podlogs.monitoring.grafana.com.spec.pipelineStages` on the cluster and
+reading [official documentation](https://grafana.com/docs/loki/latest/clients/promtail/pipelines/) how to create pipelines you can get a good understanding how to create the configuration that you need.
+
+If you don't have any needs to filter or do any custom config per application you can create a namespace wide PodLogs gatherer.
+
+```podlogs.yaml
+apiVersion: monitoring.grafana.com/v1alpha1
+kind: PodLogs
+metadata:
+  name: tenant-namespace-log
+  labels:
+    xkf.xenit.io/monitoring: tenant
+spec:
+  selector: {}
+  pipelineStages:
+    - cri: {}
+```
+
+### Tracing
+
+The tracing setup is a bit different compared to logging and metrics, instead of having some yaml file where you define the grafana agent how to gather
+data from it your application, you instead push data to the grafana agent on a specific URL within your application.
+
+Opentelemetry supports both http and grpc communication to gather traces from your application.
+
+To send http data we use `4318` and for gRPC we use `4317`.
+
+Assuming that you are using XKF the URL will be
+
+- [http://grafana-agent-traces.opentelemetry.svc.cluster.local:4318/v1/traces](http://grafana-agent-traces.opentelemetry.svc.cluster.local:4318/v1/traces)
+- [http://grafana-agent-traces.opentelemetry.svc.cluster.local:4317/v1/traces](http://grafana-agent-traces.opentelemetry.svc.cluster.local:4317/v1/traces)
+
+#### Tail-based sampling
+
+By default the grafana-agent that is deployed by XKF forwards all traces without any special config.
+This can cause high costs thanks to the amount of data that is sent. Grafana-agent have done there own solution on `probabilistic sampling`
```

Comment:
"You can configure the agent to use probabilistic sampling ..." or something similar would make an easier second sentence.